tests/test_tezos_interop.ml

```diff
@@ -0,0 +1,605 @@
+      let edpk = genesis_wallet in
+      test "to_string" (fun { expect; _ } ->
+          (expect.string (to_string edpk)).toEqual
+            "edpkvDqjL7aXdsXSiK5ChCMAfqaqmCFWCv7DaT3dK1egJt136WBiT6";
+          (expect.string (to_string TZ2_ex.pk)).toEqual
+            "sppk7aCim2kYBWGoEQGezcg8LbHi99XxrAE5dh6DDUru26pT93V2c5U";
+          (expect.string (to_string TZ3_ex.pk)).toEqual
+            "p2pk67mCLEZ2iFivprKpjugLakXfahdq9VUqmiTho1fhtv9P3AeJoc7");
+      test "of_string" (fun { expect; _ } ->
+          (expect.option
+             (of_string "edpkvDqjL7aXdsXSiK5ChCMAfqaqmCFWCv7DaT3dK1egJt136WBiT6"))
+            .toBe
+            ~equals:Key.equal (Some edpk);
+          (expect.option
+             (of_string
+                "sppk7aCim2kYBWGoEQGezcg8LbHi99XxrAE5dh6DDUru26pT93V2c5U"))
+            .toBe
+            ~equals:Key.equal (Some TZ2_ex.pk);
+          (expect.option
+             (of_string
+                "p2pk67mCLEZ2iFivprKpjugLakXfahdq9VUqmiTho1fhtv9P3AeJoc7"))
+            .toBe
+            ~equals:Key.equal (Some TZ3_ex.pk));
+      test "invalid prefix" (fun { expect; _ } ->
+          (expect.option
+             (of_string "edpuvDqjL7aXdsXSiK5ChCMAfqaqmCFWCv7DaT3dK1egJt136WBiT6"))
+            .toBeNone
+            ();
+          (expect.option
+             (of_string
+                "sddk7aCim2kYBWGoEQGezcg8LbHi99XxrAE5dh6DDUru26pT93V2c5U"))
+            .toBeNone
+            ();
+          (expect.option
+             (of_string
+                "p3pk67mCLEZ2iFivprKpjugLakXfahdq9VUqmiTho1fhtv9P3AeJoc7"))
+            .toBeNone
+            ());
+      test "invalid checksum" (fun { expect; _ } ->
+          (expect.option
+             (of_string "edpkvDqjL7aXdsXSiK5ChCMAfqaqmCFWCv7DaT3dK1egJt136WBiT5"))
+            .toBeNone
+            ();
+          (expect.option
+             (of_string
+                "sppk7dCim2kYBWGoEQGezcg8LbHi99XxrAE5dh6DDUru26pT93V2c5U"))
+            .toBeNone
+            ();
+          (expect.option
+             (of_string
+                "p2pk67mCLEz2iFivprKpjugLakXfahdq9VUqmiTho1fhtv9P3AeJoc7"))
+            .toBeNone
+            ());
+      test "invalid size" (fun { expect; _ } ->
+          (expect.option
+             (of_string "edpkvDqjL7aXdsXSiK5ChCMAfqaqmCFWCv7DaT3dK1egJt136WBiT"))
+            .toBeNone
+            ();
+          (expect.option
+             (of_string "sppk7Cim2kYBWGoEQGezcg8LbHi99XxrAE5dh6DDUru26pT93V2c5U"))
+            .toBeNone
+            ();
+          (expect.option
+             (of_string "p2pk67mCLE2iFivprKpjugLakXfahdq9VUqmiTho1fhtv9P3AeJoc7"))
+            .toBeNone
+            ()))
+let () =
+  describe "key_hash" (fun { test; _ } ->
+      let open Key_hash in
+      let tz1 = of_key genesis_wallet in
+      let tz2 = of_key TZ2_ex.pk in
+      let tz3 = of_key TZ3_ex.pk in
+      test "to_string" (fun { expect; _ } ->
+          (expect.string (to_string tz1)).toEqual
+            "tz1LzCSmZHG3jDvqxA8SG8WqbrJ9wz5eUCLC";
```

Comment:
Do you think defining the address name for this is better? for instance (you can have a better name):
`let add_1 = "tz1LzCSmZHG3jDvqxA8SG8WqbrJ9wz5eUCLC"` 
Then when we write the test below it is easier to read the code and know what we are comparing.

Reply:
this is not related to this PR